### PR TITLE
Dj/eng 592 eventing pkg testing

### PR DESCRIPTION
**Pull request by decisiveJoe**
create `rmq_adapter`, make eventing.go main export lib agnostic
unit tests for `rmq_adapter`, just cover edge cases & error conditions

### CODE REVIEW

File: eventing/rmq_adapter_test.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+func TestRmqBackend_Connect_InvalidConnectionString(t *testing.T) {
+	logger := zaptest.NewLogger(t)
+	backend := &RmqBackend{
+		connectionString: "invalid-connection-string",
+		queueName:        "test-queue",
+		logger:           logger,
+	}
+
+	err := backend.Connect()
+	assert.Error(t, err)
```
Comment (arcanez) [inline eventing/rmq_adapter_test.go:36]:
```suggestion
	require.Error(t, err)
```

File: eventing/rmq_adapter_test.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+	event := MdaiEvent{
+		Name:    "test-event",
+		Payload: "test-payload",
+		Source:  "test-source",
+		HubName: "test-hub",
+	}
+
+	err := backend.PublishMessage(event)
+	assert.Error(t, err)
```
Comment (arcanez) [inline eventing/rmq_adapter_test.go:56]:
```suggestion
	require.Error(t, err)
```

File: eventing/rmq_adapter_test.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+		queueName:        "test-queue",
+		logger:           logger,
+	}
+
+	invoker := func(event MdaiEvent) error {
+		return nil
+	}
+
+	err := backend.StartListening(invoker)
+	assert.Error(t, err)
```
Comment (arcanez) [inline eventing/rmq_adapter_test.go:73]:
```suggestion
	require.Error(t, err)
```

File: eventing/rmq_adapter_test.go [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+	assert.Contains(t, err.Error(), "connection is closed")
```
Comment (arcanez) [inline eventing/rmq_adapter_test.go:74]:
```suggestion
	assert.ErrorContains(t, err, "connection is closed")
```

File: eventing/rmq_adapter_test.go [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+	event := MdaiEvent{
+		Name:    "test-event",
+		Payload: "test-payload",
+		Source:  "test-source",
+		HubName: "test-hub",
+	}
+
+	err := backend.PublishMessage(event)
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "connection is closed")
```
Comment (arcanez) [inline eventing/rmq_adapter_test.go:57]:
```suggestion
	assert.ErrorContains(t, err, "connection is closed")
```

File: eventing/rmq_adapter_test.go [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+	logger := zaptest.NewLogger(t)
+	backend := &RmqBackend{
+		connectionString: "invalid-connection-string",
+		queueName:        "test-queue",
+		logger:           logger,
+	}
+
+	err := backend.Connect()
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "failed to dial")
```
Comment (arcanez) [inline eventing/rmq_adapter_test.go:37]:
```suggestion
	assert.ErrorContains(t, err, "failed to dial")
```

File: eventing/rmq_adapter_test.go [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+		logger:           logger,
+		isListening:      true,
+	}
+
+	invoker := func(event MdaiEvent) error {
+		return nil
+	}
+
+	err := backend.StartListening(invoker)
+	assert.NoError(t, err)
```
Comment (arcanez) [inline eventing/rmq_adapter_test.go:116]:
```suggestion
	require.NoError(t, err)
```

**Review by arcanez — CHANGES_REQUESTED** [review]
if you don't actually need the logger, `logger := zaptest.NewLogger(t)` can become `logger := zap.NewNop()`

File: eventing/rmq_adapter.go [hunk 8]
```diff
@@ -0,0 +1,305 @@
...
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	err = h.ch.PublishWithContext(ctx,
+		"",          // exchange
+		h.queueName, // routing key
+		false,       // mandatory
+		false,       // immediate
```
[the remark this answers is not in the document]
Comment (Robin-Law) [reply eventing/rmq_adapter.go:132]: I disagree; very helpful in a PR context or any other context without IDE help. 

I would honestly like to see more of them in the rest of the multilined args below.

File: eventing/rmq_adapter.go [hunk 9]
```diff
@@ -0,0 +1,305 @@
...
+	if err != nil {
+		if closeErr := conn.Close(); closeErr != nil {
+			h.logger.Error("Failed to close connection", zap.Error(closeErr))
+		}
+		return fmt.Errorf("failed to open channel: %w", err)
+	}
+
+	q, err := ch.QueueDeclare(
+		h.queueName, // name
+		true,        // durable - make queue persistent
```
[the remark this answers is not in the document]
Comment (decisiveJoe) [reply eventing/rmq_adapter.go:64]: I don't think so.

helm chart rmq values: https://github.com/DecisiveAI/mdai-helm-chart/blob/7844e846d9ef89edc25c74310db1f9144d1984ef/values.yaml#L478

[the remark this answers is not in the document]
Comment (decisiveJoe) [reply eventing/rmq_adapter.go:64]: [Looks possible](https://artifacthub.io/packages/helm/bitnami/rabbitmq#use-existing-persistentvolumeclaims), but I might need to confer with @arcanez to make sure I don't mess anything up. Will hold off on that while we're considering a shift to other lib anyway.

**Review by arcanez — APPROVED** [review]

File: eventing/eventing_mock.go
Comment (bvigar) [inline eventing/eventing_mock.go:1]: for mocking, i've really enjoyed using https://github.com/matryer/moq

you just put a `// go:generate` comment in the file with your interface, and when you run `go generate ./...` on the codebase, it creates a nice mock you can use for tests, so you don't have to worry about updating it each time you make changes to teh interface
